Agent ui

WenjiaoYue opened the pull request:
## Description

Add AgentQnA UI code

Integrate the AgentQnA UI code, along with an updated README specific to the UI, a Dockerfile for deployment, and the necessary startup scripts.

## Issues

 `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)
- [ ] Others (enhancement, documentation, validation, etc.)

## Dependencies

no

## Tests

npm  run dev

lkk12014402 commented:
https://github.com/opea-project/GenAIExamples/issues/1236

The UI is only for the ChatCompletion API and the first version is in the non-streaming form

minmin-intel reviewed (commented):
Why do we choose svelte? It seems that streamlit and gradio are more popular in ecosystem

minmin-intel commented:
Also, there a couple of warnings for the UI code...Should we address those? @lkk12014402 @WenjiaoYue

WenjiaoYue commented:
> Please update the ui usage into the readme.

Updated UI readme.

WenjiaoYue commented:
> Why do we choose svelte? It seems that streamlit and gradio are more popular in ecosystem
Streamlit and Gradio are both simpler and quicker, but they have limited flexibility and customization capabilities; Svelte is a web development front-end framework, suitable for highly customized web development, offering better extensibility and flexibility for future growth.

WenjiaoYue commented:
> Also, there a couple of warnings for the UI code...Should we address those? @lkk12014402 @WenjiaoYue

Yes, the warning has been resolved.

lkk12014402 commented:
> > Why do we choose svelte? It seems that streamlit and gradio are more popular in ecosystem
> > Streamlit and Gradio are both simpler and quicker, but they have limited flexibility and customization capabilities; Svelte is a web development front-end framework, suitable for highly customized web development, offering better extensibility and flexibility for future growth.

Additionally: 

gradio's underlying front-end framework is svelte

minmin-intel reviewed (commented):
Do we need to add a launch script for launching UI container? Also, can you add a few lines of instructions in readme on how to launch UI? @WenjiaoYue @lkk12014402

WenjiaoYue commented:
> Do we need to add a launch script for launching UI container? Also, can you add a few lines of instructions in readme on how to launch UI? @WenjiaoYue @lkk12014402

Yes, I have added the command to launch the frontend service with a Docker container in the README.

minmin-intel commented:
> > Do we need to add a launch script for launching UI container? Also, can you add a few lines of instructions in readme on how to launch UI? @WenjiaoYue @lkk12014402
> 
> Yes, I have added the command to launch the frontend service with a Docker container in the README.

@WenjiaoYue Sorry I don't see the commands...

WenjiaoYue commented:
> > > Do we need to add a launch script for launching UI container? Also, can you add a few lines of instructions in readme on how to launch UI? @WenjiaoYue @lkk12014402
> > 
> > 
> > Yes, I have added the command to launch the frontend service with a Docker container in the README.
> 
> @WenjiaoYue Sorry I don't see the commands...


The README of the UI describes two ways to start the frontend: one is by running the local command `npm run dev`, and the other is by using Docker.

![image](https://github.com/user-attachments/assets/0ac16fd8-6c64-403f-b4bc-94b8d466335f)

minmin-intel commented:
> > > > Do we need to add a launch script for launching UI container? Also, can you add a few lines of instructions in readme on how to launch UI? @WenjiaoYue @lkk12014402
> > > 
> > > 
> > > Yes, I have added the command to launch the frontend service with a Docker container in the README.
> > 
> > 
> > @WenjiaoYue Sorry I don't see the commands...
> 
> The README of the UI describes two ways to start the frontend: one is by running the local command `npm run dev`, and the other is by using Docker.
> 
> ![image](https://private-user-images.githubusercontent.com/108783334/403654101-0ac16fd8-6c64-403f-b4bc-94b8d466335f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY5OTgxMTgsIm5iZiI6MTczNjk5NzgxOCwicGF0aCI6Ii8xMDg3ODMzMzQvNDAzNjU0MTAxLTBhYzE2ZmQ4LTZjNjQtNDAzZi1iNGJjLTk0YjhkNDY2MzM1Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExNlQwMzIzMzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NmNhYTMwNGNlNTk4M2E4ZWNlOWRiMzJlODg4ZDBhMjk5YzQxYmI1YzZlMDNjMjlkMGZkOTc5Y2Y1OGM2NDM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.DuT0l5XiCwjP09tHyvSpNaREMnSNpeyzZI8L0Ceuwls)

Thanks for pointing it out. I was looking in the main readme. Can you please add instructions in the main README for UI. Maybe a sentence that links to the README in the UI folder. Thanks! @WenjiaoYue

WenjiaoYue commented:
> > > > > Do we need to add a launch script for launching UI container? Also, can you add a few lines of instructions in readme on how to launch UI? @WenjiaoYue @lkk12014402
> > > > 
> > > > 
> > > > Yes, I have added the command to launch the frontend service with a Docker container in the README.
> > > 
> > > 
> > > @WenjiaoYue Sorry I don't see the commands...
> > 
> > 
> > The README of the UI describes two ways to start the frontend: one is by running the local command `npm run dev`, and the other is by using Docker.
> > ![image](https://private-user-images.githubusercontent.com/108783334/403654101-0ac16fd8-6c64-403f-b4bc-94b8d466335f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY5OTgxMTgsIm5iZiI6MTczNjk5NzgxOCwicGF0aCI6Ii8xMDg3ODMzMzQvNDAzNjU0MTAxLTBhYzE2ZmQ4LTZjNjQtNDAzZi1iNGJjLTk0YjhkNDY2MzM1Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExNlQwMzIzMzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04NmNhYTMwNGNlNTk4M2E4ZWNlOWRiMzJlODg4ZDBhMjk5YzQxYmI1YzZlMDNjMjlkMGZkOTc5Y2Y1OGM2NDM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.DuT0l5XiCwjP09tHyvSpNaREMnSNpeyzZI8L0Ceuwls)
> 
> Thanks for pointing it out. I was looking in the main readme. Can you please add instructions in the main README for UI. Maybe a sentence that links to the README in the UI folder. Thanks! @WenjiaoYue
ok, I have added.
